java-checks/src/main/java/org/sonar/java/checks/PrintfCheck.java

```diff
@@ -121,14 +122,30 @@ private void handlePrintfFormat(MethodInvocationTree mit, String formatString, L
   private void handleMessageFormat(MethodInvocationTree mit, String formatString, List<ExpressionTree> args) {
     String newFormatString = cleanupDoubleQuote(formatString);
     Set<Integer> indexes = getMessageFormatIndexes(newFormatString);
+    List<ExpressionTree> newArgs = args;
+    if (newArgs.size() == 1) {
+      ExpressionTree firstArg = args.get(0);
```

Comment:
This is correct, but it looks strange that you check `newArgs.size()` and then access `args.get(0)`, use `newArgs.get(0)` instead.